docs/diagrams/AddNote_Sequence.puml

```diff
@@ -0,0 +1,86 @@
+@startuml
+box "Parser" #LightBlue
```

Comment:
And one more thing regarding the UML diagram there are some sections where the rectangular blocks are missing when the control is passed from one class to another.